RitAssist device tracker support

wimhaanstra opened the pull request:
## Description:


**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#5549

## Example entry for `configuration.yaml` (if applicable):
```yaml
device_tracker:
  - platform: ritassist
    client_id: [CLIENT_ID]
    client_secret: [CLIENT_SECRET]
    username: [USERNAME]
    password: [PASSWORD]
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New or updated dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [x] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

Commit:
RitAssist device tracker support

Add support for a device tracker based on the RitAssist platform.

Commit:
Fix linter errors

Commit:
Fix import order

Commit:
Place imports in functions that need them

- Place imports in functions that need them
- Add extra information to vehicle

Commit:
Include tests

Include tests
Fix compile error

Commit:
Better querystring format

Commit:
Fix linting errors

Fix linting errors

Commit:
Workaround for ci syntax error error

Commit:
Spacing

Commit:
Remove test to see if it fixes CI build

Commit:
Add some more tests

Commit:
Newline at the end of test file

Commit:
Updated tests
If no see object, skip seeing

Commit:
We really should get people some wider monitors… 79 chars on a line now.

Commit:
Remove unused import

depl0y commented on tests/components/device_tracker/test_ritassist.py:
```diff
@@ -0,0 +1,73 @@
+                     text=load_fixture('ritassist_fleet.json'))
+
+        mock_req.get('https://secure.ritassist.nl/GenericServiceJSONP.ashx',
+                     text=load_fixture('ritassist_extra_information.json'))
```
outdated

sjabby committed:
Frontend: Allow overriding default url when added to home screen (#15368)

Frontend: Allow overriding default url when added to home screen

worm-ee committed:
Add HomematicIP Cloud light power consumption and energie attributes (#15343)

* Add power consumption and energie attributes

* Fix lint

* Change attribute name and include kwh

dgomes committed:
fix camera.push API overwrite (#15334)

* fix camera.push API overwrite

* dont search in the component dictionary, but in hour own

* remove error message

* hound

NoUsername committed:
Add support for multi-channel enocean switches (D2-01-12 profile) (#14548)

starkillerOG committed:
Add sound mode support (#14910)

* Add sound mode support

* continuation line indent

* indentation

* indentation

* Remove option to configure sound_mode_dict

* Sound mode support

- removed the sound_mode_raw propertie because it was not used, (still available through self._sound_mode_raw (as device attribute for automations and diagnostics)

* Detect sound mode support from device

Removed the config option to indicate if sound mode is supported.
Added detection if sound mode is supported from the receiver itself.
Pushed denonavr library to V.0.7.4

* Pushed denonavr to v.0.7.4

iliketoprogram14 committed:
Fixed issue 15340. alexa/smart_home module can now skip properties that aren't supported in the current state, eg lowerSetpoint in Heat mode or targetSetpoint in Eco mode for Nest devices. (#15352)

fabaff committed:
Efergy (#15380)

* Update format

* Use string formatting

balloob committed:
Use IndieAuth for client ID (#15369)

* Use IndieAuth for client ID

* Lint

* Lint & Fix tests

* Allow local IP addresses

* Update comment

balloob committed:
Bump frontend to 20180709.0

scop committed:
Add httplib2 to h.c.google requirements (#15385)

scop committed:
Remove some unneeded pylint import-error disables (#15386)

danielperna84 committed:
Update ha-philipsjs to 0.0.5 (#15378)

* Update requirements_all.txt

* Update philips_js.py

hanzoh committed:
Add new voices to Amazon Polly (#15320)

ludeeus committed:
Add Cloudflare DNS component. (#15388)

* Add Cloudflare DNS component

* Removed man

* Update .coveragerc

* Update cloudflare.py

* Update cloudflare.py

* Changed records to be required

* Fix typos, update order and other minor changes

robmarkcole committed:
Add Facebox teach service (#14998)

* Adds service

* Address pylint

* Update facebox.py

* patch tests

* Update facebox.py

* Update test_facebox.py

* Update facebox.py

* Update facebox.py

* Update facebox.py

* Update test_facebox.py

* Update test_facebox.py

* Update facebox.py

* Update facebox.py

* Update facebox.py

* Update facebox.py

* Adds total_matched_faces

* Update test_facebox.py

* Update facebox.py

* Update test_facebox.py

* Update test_facebox.py

* Remove fixtures

Removes the fixtures which were causing `setup` to fail, replace with `@patch`

* Fix teach service test and lint issues

balloob committed:
Expire auth code after 10 minutes (#15381)

balloob committed:
Bump frontend to 20180710.0

glpatcern committed:
Improve NetAtmo sensors update logic (#14866)

* Added a "last update" sensor that could be used by automations + cosmetic changes

* Improved the update logic of sensor data

The platform is now continuously adjusting the refresh interval
in order to synchronize with the expected next update from the
NetAtmo cloud. This significantly improves reaction time of
automations while keeping the refresh time to the recommended
value (10 minutes).

* Linting

* Incorporated the advanced Throttle class to support adaptive
throttling, as opposed to integrating it in the core framework.

Following code review, it was suggested to implement the
specialised Throttle class in this platform instead of making a
change in the general util package. Except that the required change
(about 4 LoC) is part of the only relevant piece of code of that
class, therefore this commit includes a full copy of the Throttle
class from homeassistant.util, plus the extra feature to support
adaptive throttling.

* Cosmetic changes on the introduced "last updated" sensor

* Alternate implementation for the adaptive throttling

Ensure the updates from the cloud are throttled and adapted to the
last update time provided by NetAtmo, without using the Throttle
decorator. Similar logic and similar usage of a lock to protect
the execution of the remote update.

* Linting

ludeeus committed:
removed unused return (#15402)

amelchio committed:
Fix confused brightness of xiaomi_aqara gateway light (#15314)

pschmitt committed:
Fix liveboxplaytv empty channel list (#15404)

balloob committed:
Fix credentials lookup (#15409)

tchellomello committed:
Change Ring binary_sensor frequency polling to avoid rate limit exceeded errors (#15414)

huangyupeng committed:
Add Tuya component and switch support (#15399)

* support for tuya platform

* support tuya platform

* lint fix

* change dependency

* add tuya platform support

* remove tuya platform except switch. fix code as required

* fix the code as review required

* fix as required

* fix a mistake

balloob committed:
Merge branch 'master' into dev

danielperna84 committed:
Fix HomeMatic variables (#15417)

* Update __init__.py

* Update requirements_all.txt

mjg59 committed:
Ignore some HomeKit devices (#15316)

There are some devices that speak HomeKit that we shouldn't expose. Some
bridges (such as the Hue) provide reduced functionality over HomeKit and
have a functional native API, so should be ignored. We also shouldn't
offer to configure the built-in Home Assistant HomeKit bridge.

amelchio committed:
Make LimitlessLED color/temperature attributes mutually exclusive (#15298)

worm-ee committed:
Add HomematicIP alarm control panel (#15342)

* Add HomematicIP security zone

* Update access point tests

* Fix state if not armed and coments

* Add comment for the empty state_attributes

* Fix comment

* Fix spelling

awarecan committed:
Include request.path in legacy api password warning message (#15438)

andrey-git committed:
Add python 3.8-dev to travis and tox (#15347)

* Add Python 3.8-dev tox tests.

* Allow failures on 3.8-dev

* Allow failures on 3.8-dev take2

* Only run on pushes to dev

balloob committed:
Reorg auth (#15443)

andrey-git committed:
Make typing checks more strict (#14429)

## Description:

Make typing checks more strict: add `--strict-optional` flag that forbids implicit None return type. This flag will become default in the next version of mypy (0.600)

Add `homeassistant/util/` to checked dirs.

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**

scop committed:
upgrade-mypy (#14904)

* Upgrade mypy to 0.600

* Upgrade mypy to 0.610

* Typing improvements

* remove unneeded or

* remove merge artifact

* Update loader.py

balloob committed:
Fix comment formatting (#15447)

balloob committed:
User management (#15420)

* User management

* Lint

* Fix dict

* Reuse data instance

* OrderedDict all the way

balloob committed:
Bump frontend to 20180713.0

andrey-git committed:
More typing (#15449)

## Description:

More typing improvements.

Switch to using `mypy.ini` for flexibility

Add `warn_return_any` check except in `homeassistant.util.yaml` that does typing hacks. Fix some type annotations as resulting from this check and ignore others were fixing is hard.

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**

fabaff committed:
Catch the ValueError if the bulb was in the wrong mode (#15434)

fabaff committed:
Upgrade keyring to 13.2.1 (#15453)

balloob committed:
Fix formatting pylint comments in test (#15450)

worm-ee committed:
Add HomematicIP Cloud dimmer light device (#15456)

* Add dimmable light device

* Add imports

* Fix float and int conversion

jantman committed:
Fix ZWave RGBW lights not producing color without explicit white_value (#15412)

* Fix ZWave RGBW lights not producing color without explicit white_value (#13930)

* simplify conditional in previous commit (#13930)

* ZwaveColorLight - only zero _white if white_value not specified in call (#13930)

danielperna84 committed:
Add IPPassageSensor (HmIP-SPDR) (#15458)

teharris1 committed:
Implement is_on (#15459)

* Implement is_on

* Remove var

scop committed:
Remove unnecessary executable permissions (#15469)

huangyupeng committed:
Add Tuya light platform (#15444)

* add tuya light platform

* fix as review required

worm-ee committed:
Update homematicip_cloud with enum states (#15460)

* Update to next version with enum states

* Change to generic dimmer class

* Update of requirement files

* Update to hmip lib to v0.9.7

* Missing update of requirements files

* Cleanup of icon properties

balloob committed:
Add user via cmd line creates owner (#15470)

* Add user via cmd line creates owner

* Ensure access tokens are not verified for inactive users

* Stale print

* Lint

andrey-git committed:
Switch to own packaged version of pygtfs (#15040)

balloob committed:
Aware comments (#15480)

* Make sure we cannot deactivate the owner

* Use different error code when trying to fetch token for inactive user

amelchio committed:
Fix flux_led turning on with color or effect (#15472)

amelchio committed:
Update limitlessled to 1.1.2 (#15481)

balloob committed:
Bump frontend to 20180716.0

balloob committed:
Version bump to 0.74.0.dev0

balloob committed:
Bumped version to 0.74.0b0

balloob committed:
Version bump to 0.75.0.dev0

balloob committed:
Extract SSL context creation to helper (#15483)

* Extract SSL context creation to helper

* Lint

balloob committed:
0.73.2 - security release (#15494)

* Extract SSL context creation to helper (#15483)

* Extract SSL context creation to helper

* Lint

* Bumped version to 0.73.2

balloob committed:
Merge branch 'master' into rc

balloob committed:
Bumped version to 0.74.0b1

squirtbrnr committed:
Delay setup of waze travel time component (#15455)

* delay setup of component

Copied the necessary lines of code from the google travel time component to fix the setup delay in waze travel time component.  Previously it was only watching the homeassistant start event on the bus, but doing nothing with it.

* Update waze_travel_time.py

* Update waze_travel_time.py

lukiffer committed:
Bump pyarlo==0.2.0, fixes #15486 (#15503)

balloob committed:
Add current user WS command (#15485)

balloob committed:
Add current user WS command (#15485)

mjg59 committed:
Update HomeKit module code (#15502)

This fixes a bunch of bugs, including issues with concurrency in devices
that present multiple accessories, devices that insist on the TLV entries
being in the order that Apple use, and handling devices that send headers
and data in separate chunks. This should improve compatibility with
a whole bunch of HomeKit devices.

mjg59 committed:
Update HomeKit module code (#15502)

This fixes a bunch of bugs, including issues with concurrency in devices
that present multiple accessories, devices that insist on the TLV entries
being in the order that Apple use, and handling devices that send headers
and data in separate chunks. This should improve compatibility with
a whole bunch of HomeKit devices.

balloob committed:
Add onboarding support (#15492)

* Add onboarding support

* Lint

* Address comments

* Mark user step as done if owner user already created

balloob committed:
Add onboarding support (#15492)

* Add onboarding support

* Lint

* Address comments

* Mark user step as done if owner user already created

balloob committed:
Bump frontend to 20180717.0

balloob committed:
Bump frontend to 20180717.0

balloob committed:
Bumped version to 0.74.0b2

scop committed:
Pylint 2 fixes (#15487)

* pylint 2 inline disable syntax fixes

* pylint 2 logging-not-lazy fixes

* pylint 2 consider-using-in fixes

* Revert pylint 2 inline disable syntax fixes addressing unused-imports

Will have a go at removing more unused imports altogether first.

awarecan committed:
Disallow use insecure_example auth provider in configuration.yml (#15504)

* Disallow use insecure_example auth provider in configuration.yml

* Add unit test for auth provider config validate

awarecan committed:
Disallow use insecure_example auth provider in configuration.yml (#15504)

* Disallow use insecure_example auth provider in configuration.yml

* Add unit test for auth provider config validate

huangyupeng committed:
Add Tuya climate platform (#15500)

* Add Tuya climate platform

* fix as review required

* fix as review required

JackNova committed:
Support latest tplink Archer D9 Firmware version / Device Scanner (#15356)

* Support latest tplink Archer D9 Firmware version / Device Scanner

* tplink integration on pypi package

* initialize the client only once

* remove unnecessary instance attributes

mu4yu3 committed:
Customizable command timeout (#15442)

* Customizable command timeout

* Change string to int

* update the tests. Do the same thing on the binary_sensor.command_line.

andrey-git committed:
Add check_untyped_defs (#15510)

* Add check_untyped_defs

* Change to regular if-else

MartinHjelmare reviewed (changes requested):
The device interface code should be extracted into separate package and published on pypi.

https://developers.home-assistant.io/docs/en/creating_platform_code_review.html#6-communication-with-devices-services

scop committed:
Remove some unused imports (#15529)

balloob committed:
Decouple emulated hue from http server (#15530)

balloob committed:
Decouple emulated hue from http server (#15530)

pvizeli committed:
Moon translate (#15498)

* Translate moon

* Create strings.moon.json

* Update moon.py

* Update strings.moon.json

* Update test_moon.py

scop committed:
More pylint 2 fixes (#15516)

* Pylint 2 useless-import-alias fixes

* Pylint 2 chained-comparison fixes

* Pylint 2 consider-using-get fixes

* Pylint 2 len-as-condition fixes

andrey-git committed:
Make sure that only pypi dependencies are used (#15490)

amelchio committed:
RFC: Only use supported light properties (#15484)

* Only use supported light properties

* Fix tests

worm-ee committed:
Cleanup of HomematicIP Cloud code (#15475)

* Check if device supports lowBat and shows it only if battery is low

* Show empty battery icon if lowBat is true

* Default return None

* Sabotage attribute and icon if device has this feature

* Bug fix and cleanup

* Use dedicated function for security state

* Cleanup of sensor attributes and icons

* Empty

dshokouhi committed:
Implement locate service for neato (#15467)

* Implement locate service for neato

* Hound

turbokongen committed:
Make RS room thermostat discoverable (#15451)

* Make RS room thermostat discoverable

* Reversed generic type name

fucm committed:
Add support for Tahoma Soke Sensor (#15441)

teharris1 committed:
Mini-Remote events (#15523)

* Add event handler to capture binary sensor on messages

* Log event trigger

* Log event firing

* Capture platform correctly

* Fix test for platform eq binary_sensor

* Create sensor events

* Add light and battery sensors

* Bump insteonplm version to 0.11.6

* Fix naming of BUTTON_PRESSED_STATE_NAME

* Fix naming of fire event methods

* Add logging

* Add DOMAIN definition

* Get state name from plm.devices

* Remove stale reference to button ID

* Fix reference to state name

* Remove incorrect ref to self

* Log remote button pressed event

* Change mode to button_mode and fix values to array

* Rename CONF_MODE to CONF_BUTTON_MODE

* Log platform create with mode

* Properly assign button_mode to track mode

* Implement is_on

* Change mini-remotes to events only

* Remove button_mode config option

* Fix reference to _fire_button_on_off_event

* Bump insteon version to 0.11.7

* Flake8 clean up

* Flake8 cleanup

* Use % format in logging per pylint

* Code review updates

* Resolve conflict

* Lint

gieljnssns committed:
Update pyatmo (#15540)

balloob committed:
Bump frontend to 20180718.0

balloob committed:
Bump frontend to 20180718.0

balloob committed:
Bumped version to 0.74.0b3

balloob committed:
Don't be so strict client-side (#15546)

balloob committed:
Don't be so strict client-side (#15546)

quthla committed:
Update reading when device is added (#15548)

balloob committed:
Allow CORS requests to token endpoint (#15519)

* Allow CORS requests to token endpoint

* Tests

* Fuck emulated hue

* Clean up

* Only cors existing methods

balloob committed:
Allow CORS requests to token endpoint (#15519)

* Allow CORS requests to token endpoint

* Tests

* Fuck emulated hue

* Clean up

* Only cors existing methods

balloob committed:
Update config entry id in entity registry (#15531)

jeradM committed:
Added WS endpoint for changing homeassistant password. (#15527)

* Added WS endpoint for changing homeassistant password.

* Remove change password helper. Don't require current password.

* Restore current password verification.

* Added tests.

* Use correct send method

jeradM committed:
Added WS endpoint for changing homeassistant password. (#15527)

* Added WS endpoint for changing homeassistant password.

* Remove change password helper. Don't require current password.

* Restore current password verification.

* Added tests.

* Use correct send method

balloob committed:
Bump frontend to 20180719.0

balloob committed:
Bump frontend to 20180719.0

balloob committed:
Bumped version to 0.74.0b4

Commit:
Upgrade to simplisafe-python v2 to use new SimpliSafe API (#15542)

* Upgrade to simplisafe-python v2 to use new SimpliSafe API

balloob committed:
Allow auth providers to influence is_active (#15557)

* Allow auth providers to influence is_active

* Fix auth script test

balloob committed:
Allow auth providers to influence is_active (#15557)

* Allow auth providers to influence is_active

* Fix auth script test

MartinHjelmare committed:
Upgrade pymysensors to 0.16.0 (#15554)

MartinHjelmare committed:
Upgrade pymysensors to 0.16.0 (#15554)

balloob committed:
Remove relative time from state machine (#15560)

balloob committed:
Remove relative time from state machine (#15560)

scop committed:
More pylint 2 fixes (#15565)

## Description:

More fixes flagged by pylint 2 that don't hurt to have before the actual pylint 2 upgrade (which I'll submit soon).

## Checklist:
  - [ ] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**

huangyupeng committed:
Add Tuya fan support (#15525)

* Add Tuya fan platform

* Add Tuya fan platform

* fix as review required

awarecan committed:
Reset failed login attempts counter when login success (#15564)

awarecan committed:
Reset failed login attempts counter when login success (#15564)

balloob committed:
Update the frontend to 20180720.0

balloob committed:
Update the frontend to 20180720.0

balloob committed:
Bumped version to 0.74.0

rytilahti committed:
light.tplink: initialize min & max mireds only once, avoid i/o outside update (#15571)

* light.tplink: initialize min & max mireds only once, avoid i/o outside update

* revert the index change

* fix indent, sorry for overwriting your fix, balloob

rytilahti committed:
light.tplink: initialize min & max mireds only once, avoid i/o outside update (#15571)

* light.tplink: initialize min & max mireds only once, avoid i/o outside update

* revert the index change

* fix indent, sorry for overwriting your fix, balloob

balloob committed:
Merge pull request #15570 from home-assistant/rc

0.74

azogue committed:
fix aiohttp InvalidURL exception when fetching media player image (#15572)

* fix aiohttp InvalidURL exception when fetching media player image

The first call for the HA proxy (`/api/media_player_proxy/media_player.kodi?token=...&cache=...`)
is receiving relative urls that are failing, this is a simple fix to precede the base_url when hostname is None.

* fix import location and sort stdlib imports

balloob committed:
Merge branch 'master' into dev

PrimusNZ committed:
Allow MQTT Switch to have an optional state configuration (#15430)

Switches by default use the payload_on and payload_off configuration parameters to specify both the payload the switch should send for a state but also what will be returned for the current state - which isnt always the same
As a toggle switch might always send an ON or TOGGLE to toggle the switch, but still receive an ON or an OFF for the state topic - This change allows for splitting them apart

amelchio committed:
Add unique_id to netgear_lte sensors (#15584)

digiblur committed:
Add relay addr & chan config to alarmdecoder zones (#15242)

Add relay addr & chan config to alarmdecoder zones

danielperna84 committed:
Update pyhomematic, fixes #15054, #15190 (#15603)

amelchio committed:
Throttle unavailability warnings for tplink light/switch (#15591)

awarecan committed:
Frontend component should auto load auth coomponent (#15606)

cdce8p committed:
Bugfix HomeKit name and serial_number (#15600)

* Bugfix HomeKit name and serial_number

* Revert serial_number changes

balloob committed:
Remove entity picture of Tuya entity (#15611)

scop committed:
Upgrade mypy to 0.620 (#15612)

scop committed:
Pylint cleanups (#15626)

* Pylint 2 no-else-return fixes

* Remove unneeded abstract-class-not-used pylint disable

andrey-git committed:
Add typing to homeassistant/*.py and homeassistant/util/ (#15569)

* Add typing to homeassistant/*.py and homeassistant/util/

* Fix wrong merge

* Restore iterable in OrderedSet

* Fix tests

amelchio committed:
Use case insensitive comparison for Sonos model check (#15604)

amelchio committed:
Refresh Sonos source list on changes (#15605)

sherazlodhi committed:
The tense is wrong (#15614)

Instead of spent, we should be using spend :)

starkillerOG committed:
Suppress error between 00:00 and 01:00 (#15555)

* Suppress error between 00:00 and 01:00

Suppress an error that often occers between 00:00 and 01:00 CE(S)T during that time, probably because buienradar.nl is then updating its forcast for the next day. The API does not always work between these times (in the middle of the night).

* white space & import

* unnecessary brackets

balloob committed:
Use async_create_task (#15633)

* Use async_create_task

* Fix test

balloob committed:
Allow system users to refresh tokens (#15574)

balloob committed:
WIP: Hass.io sent token to supervisor (#15536)

Hass.io sent token to supervisor

pvizeli committed:
Fix aiohttp connection reset errors (#15577)

* Fix aiohttp connection reset errors

* Update aiohttp_client.py

* Update aiohttp_client.py

* Update __init__.py

* Update mjpeg.py

* Update mjpeg.py

* Update ffmpeg.py

* Update ffmpeg.py

* Update ffmpeg.py

* Update proxy.py

* Update __init__.py

* Update aiohttp_client.py

* Update aiohttp_client.py

* Update proxy.py

* Update proxy.py

* Fix await inside coroutine

* Fix async syntax

* Lint

balloob committed:
Cast/Sonos: create config entry if manually configured (#15630)

* Cast/Sonos: create config entry if manually configured

* Add test for helper

turbokongen committed:
Add ability to set Zwave protection commandclass (#15390)

* Add API for protection commandclass

* Adjusting

* tests

* Spelling

* Missed flake8

* Period

* spelling

* Review changes

* removing additional .keys()

* period

* Move i/o out into executor pool

* Move i/o out into executor pool

* Forgot get method

* Do it right... I feel stupid

* Long lines

* Merging

dshokouhi committed:
Update Neato library to allow for dynamic endpoints (#15639)

wingy3181 committed:
Fix typo `asayn_init` instead of `async_init` (#15645)

awarecan committed:
Add ipban for failed login attempt in new login flow (#15551)

* Add ipban for failed login attempt in new login flow

* Address review comment

* Use decorator to clean up code

huangyupeng committed:
Add Tuya cover and scene platform (#15587)

* Add Tuya cover platform

* Add Tuya cover and scene

* fix description

* remove scene default method

visibilityspots committed:
Update smappy library version (#15636)

Adding latest smappy lib version

Updated smappy library version

balloob committed:
Allow changing entity ID (#15637)

* Allow changing entity ID

* Add support to websocket command

* Address comments

* Error handling

balloob committed:
Bump frontend to 20180724.0

glpatcern committed:
Downgrade netatmo warning log to info (#15652)

awarecan committed:
Add turn_on/off service to camera (#15051)

* Add turn_on/off to camera

* Add turn_on/off supported features to camera.

Add turn_on/off service implementation to camera, add turn_on/off
 supported features and services to Demo camera.

* Add camera supported_features tests

* Resolve code review comment

* Fix unit test

* Use async_add_executor_job

* Address review comment, change DemoCamera to local push

* Rewrite tests/components/camera/test_demo

* raise HTTPError instead return response

kalmard0 committed:
Allow defining default turn-on values for lights in the profiles file. (#15493)

* Allow defining default turn-on values for lights in the profiles file.

* Mock out file operations in unit test.

* Fix unit test flakiness.

* Avoid unnecessary copy

jeradM committed:
Added user credentials to current_user ws endpoint. (#15558)

* Added user credentials to current_user ws endpoint.

* Comments. Added another test.

* Return list of credentials.

scop committed:
Import collections abstract base classes from collections.abc (#15649)

Accessing them directly through collections is deprecated since 3.7, and
will no longer work in 3.8.

scop committed:
Auth typing improvements (#15640)

* Always return bytes from auth.providers.homeassistant.hash_password

Good for interface cleanliness, typing etc.

* Add some homeassistant auth provider type annotations

balloob committed:
Fix CORS duplicate registration (#15670)

peternijssen committed:
Add spider thermostat (#15499)

* add spider thermostats

* Added load_platform. Added operation dictionary. Minor improvements

* loop over spider components for load_platform

* added empty dict to load_platform. changed add_devices

* moved logic to the API

* fix requirements_all.txt

* minor code improvements

eavanvalkenburg committed:
Add Brunt Cover Device (#15653)

* New Brunt Branch

* Some small changes and updates based on review.

scop committed:
Pylint 2 useless-return fixes (#15677)

awarecan committed:
Allow Nest Cam turn on/off (#15681)

* Allow Nest Cam turn on/off

* Don't raise Error

* Remove unnecessary state update

scop committed:
Upgrade pylint to 2.0.1 (#15683)

* Upgrade pylint to 2.0.1

* Pylint 2 bad-whitespace fix

* Pylint 2 possibly-unused-variable fixes

* Pylint 2 try-except-raise fixes

* Disable pylint fixme for todoist for now

https://github.com/PyCQA/pylint/pull/2320

* Disable pylint 2 useless-return for now

https://github.com/PyCQA/pylint/issues/2300

* Disable pylint 2 invalid-name for type variables for now

https://github.com/PyCQA/pylint/issues/1290

* Disable pylint 2 not-an-iterable for now

https://github.com/PyCQA/pylint/issues/2311

* Pylint 2 unsubscriptable-object workarounds

* Disable intentional pylint 2 assignment-from-nones

* Disable pylint 2 unsupported-membership-test apparent false positives

* Disable pylint 2 assignment-from-no-return apparent false positives

* Disable pylint 2 comparison-with-callable false positives

https://github.com/PyCQA/pylint/issues/2306

balloob committed:
Bump frontend to 20180726.0

balloob committed:
Fix frontend tests

bachya committed:
Fixes a bug with showing a subset of Pollen index conditions (#15694)

peternijssen committed:
Add spider power plug component (#15682)

* Add spider power plug component

* rounding down the numbers

* ability to throttle the API

* updated to the lastest api

* resolved an issue within the API

juhaniemi committed:
Add support for P5 FutureNow light platform (#15662)

* Added support for FutureNow light platform and relay/dimmer units

* Pinned specific version for requirement

* Added support for FutureNow light platform and relay/dimmer units

* Added futurenow.py to .coveragerc.

* Minor fixes and enhancements as requested in the code review.

* Minor fixes and enhancements as requested in the code review.

* Use device_config's value directly as it's validated as boolean.

* Simplify state check.

* Fixed brightness update that was broken in previous commit.

awarecan committed:
Only log change to use access token warning once (#15690)

rorr73 committed:
Add support for alarm_control_panel to MQTT Discovery. (#15689)

jcconnell committed:
Add Magicseaweed API support (#15132)

* Added support for magicseaweed surf forecasting

* Added support for magicseaweed surf forecasting

* Added support for magicseaweed surf forecasting

* Incorporate @bachya requested changes.

* Adding support for magicseaweed package.

* Run tests and fix errors.

* Incorporate @balloob requested changes.

* Attempt to fix pylint error e1101.

* Two spaces before inline comments

* Add @MartinHjelmare & @balloob requested changes.

* Remove MagicSeaweedData object inheritance.

* Fix variable logic.

alanfischer committed:
Upgrade pyvera to 0.2.44 (#15708)

dingusdk committed:
Remove IHC XML Element from discovery data (#15719)

balloob committed:
Context (#15674)

* Add context

* Add context to switch/light services

* Test set_state API

* Lint

* Fix tests

* Do not include context yet in comparison

* Do not pass in loop

* Fix Z-Wave tests

* Add websocket test without user

awarecan committed:
Break up components/auth (#15713)

jkeljo committed:
Add a component for Sisyphus Kinetic Art Tables (#14472)

* Add a component for Sisyphus Kinetic Art Tables

The [Sisyphus Kinetic Art Table](https://sisyphus-industries.com/) uses a
steel ball to draw intricate patterns in sand, thrown into sharp relief by a
ring of LED lights around the outside.

This component enables basic control of these tables through Home Assistant.

* Fix lints

* Docstrings, other lints

* More lints

* Yet more.

* Feedback

* Lint

* Missed one piece of feedback

* - Use async_added_to_hass in media player
- async_schedule_update_ha_state in listeners
- constants for supported features
- subscripting for required keys
- asyncio.wait
- update to sisyphus-control with passed-in session

* Update requirements

* lint

Commit:
Command Line Sensor - json_attributes (#15679)

* Add tests to command_line for json_attrs

* Add json_attrs to command_line

* Remove whitespace on blank line

* Stick to <80 row length

* Use collections.Mapping, not dict

* Rename *attrs to *attributes

* Remove extraneous + for string concat

* Test multiple keys

* Add test

Makes sure the sensor's attributes don't contain a value for a missing key,
even if we want that key.

* Test that unwanted keys are skipped

* Remove additional log line

* Update tests for log changes

* Fix ordering

fabaff committed:
Upgrade youtube_dl to 2018.07.21 (#15718)

fabaff committed:
Upgrade numpy to 1.15.0 (#15722)

peternijssen committed:
Upgrade spiderpy to 1.2.0 (#15729)

fabaff committed:
Upgrade beautifulsoup4 to 4.6.1 (#15727)

fabaff committed:
Upgrade mutagen to 1.41.0 (#15739)

fabaff committed:
Upgrade sqlalchemy to 1.2.10 (#15737)

fabaff committed:
Upgrade voluptuous to 0.11.3 (#15735)

fabaff committed:
Upgrade youtube_dl to 2018.07.29 (#15734)

fabaff committed:
Upgrade pyowm to 2.9.0 (#15736)

fabaff committed:
Various updates (#15738)

starkillerOG committed:
Denonavr version push to 0.7.5 (#15743)

* Version push to 0.7.5

Improve logger warning

* Denonavr v.0.7.5

juhaniemi committed:
Make FutureNow light remember last brightness when turning on (#15733)

* Remember last brightness value and use it on turn_on()

* Pyfnip-0.2 now returns state reliably, no manual changes needed.

* Split too long line of code

* Updated pyfnip library version

shoejosh committed:
Add Genie Aladdin Connect cover component (#15699)

* Add Genie Aladdin Connect cover component

* Fix lines being too long

* Fix issues found in review

* remove Unknown state, use None instead

* Fixed requirements_all

DavidMStraub committed:
Add mvglive option to store multiple departures in attributes (#15454)

* MVG Live sensor: add option to store multiple departures in attributes

* Fix lint error

* mvglive: take into account timeoffset in API call

* Prevent exception if departure list is empty

* Rename state_attributes -> device_state_attributes

danielfaulknor committed:
Add other wemo motion sensor identifier (#15627)

* Add other motion sensor identifier

* Fix order

awarecan committed:
Remove self type hints (#15732)

* Remove self type hints

* Lint

superpuffin committed:
Upgrade Adafruit-DHT to 1.3.3 (#15706)

* Change to newer pip package

The package Adafruit_Python_DHT==1.3.2 was broken and would not install, breaking DHT sensor support in Home assistant. It has since been fixed in Adafruit-DHT==1.3.3.

See: https://github.com/adafruit/Adafruit_Python_DHT/issues/99

* Update requirements_all.txt

New or updated dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.

* Comment out Adafruit-DHT

Adafruit_Python_DHT changed name to Adafruit-DHT, which still need pyx support breaking our CI, need to be comment out.

* Update requirements_all.txt

rytilahti committed:
Add kodi unique id based on discovery (#15093)

* kodi add unique id based on discovery

* initialize unique_id to None

* use netdisco-extracted mac_address

* use an uuid instead of mac for real uniqueness

* add missing docstring

* verify that there is no entity already for the given unique id

* whitespace fix

Commit:
Fix inheritance issues

Commit:
Fix inheritance issues

Commit:
Fix unused imports

Commit:
Fix unused imports

depl0y commented:
@MartinHjelmare The changes to move API calls to a separate package came after the commits took place. Also, I never did this and seems like a huge step for a lot of developers to actually support this?

MartinHjelmare commented:
What changes are you referring to?

Having the device and service interface code in standalone libraries is one of the main foundations of home assistant. We won't change that.

I can recommend the cookie cutter project that helps create a python package skeleton to get started quickly.

https://github.com/audreyr/cookiecutter

bachya committed:
Better handling of Yi camera being disconnected (#15754)

* Better handling of Yi camera being disconnected

* Handling video processing as well

* Cleanup

* Member-requested changes

* Member-requested changes

awarecan committed:
Decouple login flow view and data entry flow view (#15715)

foxel committed:
Fixed NDMS for latest firmware (#15511)

* Fixed NDMS for latest firmware.
Now using telnet instead of Web Interface

* Using external library for NDMS interactions

* updated requirements_all

* renamed `mac` to `device` back

* Using generators for name and attributes fetching

Commit:
Move API requests to external library

Commit:
Move API requests to external library

Commit:
Fix linting issues

Commit:
Fix linting issues

Commit:
Trailing whitespace

Commit:
Trailing whitespace

depl0y commented:
@MartinHjelmare Moved API stuff to separate module

MartinHjelmare commented on tests/components/device_tracker/test_ritassist.py:
```diff
@@ -0,0 +1,60 @@
+        """Test for successfully setting up the platform."""
+        self._setup_mock_requests(mock_req)
+
+        scanner = RitAssistDeviceScanner(self.hass, self.config, None, None)
```
Please use the home assistant core to set up the device tracker component and ritassist platform. Then use the state machine to get the state and assert that it's correct.

Use the `setup_component` function from homeassistant.setup.py.

depl0y replied:
Using `setup_component` causes a couple of errors, mainly:

```
ERROR:homeassistant.setup:Error during setup of component device_tracker
Traceback (most recent call last):
  File "/Volumes/Projects/forks/home-assistant/homeassistant/setup.py", line 146, in _async_setup_component
    component.setup, hass, processed_config)  # type: ignore
AttributeError: module 'custom_components.device_tracker' has no attribute 'setup'
INFO:homeassistant.core:Bus:Handling <Event call_service[L]: domain=persistent_notification, service=create, service_data=title=Invalid config, message=The following components and platforms could not be set up:

 - [device-tracker](https://home-assistant.io/components/device_tracker/)
```

depl0y replied:
Moving tests to external module. Not much to test left in this device tracker.

MartinHjelmare replied:
Are you using a custom component?

MartinHjelmare reviewed (commented):
Good!

See comments below. Also make sure you run `script/gen_requirements.py` to update the requirements file.

andrey-git committed:
Add disallow_untyped_calls to mypy check. (#15661)

* Add disallow_untyped_calls to mypy check.

* Fix generator

turbokongen committed:
Update pyozw to 0.4.9 (#15758)

* update pyozw to 0.4.8

* add requirements_all.txt

* use 0.4.9

Commit:
Remove authentication caching, can be done by external module
Clean up unused variables

Commit:
Remove authentication caching, can be done by external module
Clean up unused variables

depl0y commented on homeassistant/components/device_tracker/ritassist.py:
```diff
@@ -0,0 +1,117 @@
+                              icon='mdi:car')
+
+        except requests.exceptions.ConnectionError:
+            _LOGGER.error('ConnectionError: Could not connect to RitAssist')
```
Credentials are checked when the request is performed. If the validation fails, it can try again  later. If absolutely necessary, I will update it, but authentication can also fail when the RitAssist/FleetGO platform is updating.

MartinHjelmare replied:
If the credentials are bad, the platform will never work until the user fixes it. We should let the user know that, and fail the setup.

depl0y commented on homeassistant/components/device_tracker/ritassist.py:
```diff
@@ -0,0 +1,117 @@
+
+    def save_authentication(self, filename):
+        """Save the authentication information to a file for caching."""
+        from homeassistant.util.json import save_json
```
Removed, caching of credentials can be done in the external module.

depl0y commented on homeassistant/components/device_tracker/ritassist.py:
```diff
@@ -0,0 +1,117 @@
+
+    def load_authentication(self, filename):
+        """Load the authentication information from a file for caching."""
+        from homeassistant.util.json import load_json
```
Removed, caching of credentials can be done in the external module.

depl0y commented on homeassistant/components/device_tracker/ritassist.py:
```diff
@@ -0,0 +1,117 @@
+
+    def _refresh(self) -> None:
+        """Refresh device information from the platform."""
+        import requests
```
Not sure what you mean by this. Removed a truckload of code, maybe it's better now?

depl0y commented on homeassistant/components/device_tracker/ritassist.py:
```diff
@@ -0,0 +1,117 @@
+import logging
+import voluptuous as vol
+
+from ritassist import (API, Authentication)
```
Updated

depl0y commented on homeassistant/components/device_tracker/ritassist.py:
```diff
@@ -0,0 +1,117 @@
+        self._discovery_info = discovery_info
+        self._hass = hass
+        self._devices = []
+        self._config = config
```
Updated

depl0y commented on homeassistant/components/device_tracker/ritassist.py:
```diff
@@ -0,0 +1,117 @@
+
+        self._discovery_info = discovery_info
+        self._hass = hass
+        self._devices = []
```
Updated

depl0y commented on homeassistant/components/device_tracker/ritassist.py:
```diff
@@ -0,0 +1,117 @@
+        except requests.exceptions.ConnectionError:
+            _LOGGER.error('ConnectionError: Could not connect to RitAssist')
+
+    def save_authentication(self, filename):
```
Removed, caching of credentials can be done in the external module.

depl0y commented on homeassistant/components/device_tracker/ritassist.py:
```diff
@@ -0,0 +1,117 @@
+        """Initialize RitAssistDeviceScanner."""
+        from homeassistant.helpers.event import track_utc_time_change
+
+        self._discovery_info = discovery_info
```
Updated

depl0y commented on homeassistant/components/device_tracker/ritassist.py:
```diff
@@ -0,0 +1,117 @@
+        from homeassistant.helpers.event import track_utc_time_change
+
+        self._discovery_info = discovery_info
+        self._hass = hass
```
Updated

Commit:
Update lint issue with long line

Commit:
Update lint issue with long line

Commit:
Updated my IDE to support Pylint 😉

Commit:
Updated my IDE to support Pylint 😉

depl0y commented on homeassistant/components/device_tracker/ritassist.py:
```diff
@@ -0,0 +1,117 @@
+https://home-assistant.io/components/device_tracker.ritassist/
+"""
+import logging
+import voluptuous as vol
```
Ypdated.

MartinHjelmare replied:
Add a blank line between logging and voluptuous.

Commit:
Move tests to external module

Commit:
Move tests to external module

fabaff committed:
Convert wind speed to km/h (fixes #15710) (#15740)

* Convert wind speed to km/h (fixes #15710)

* Round speed

priiduonu committed:
Round precipitation forecast to 1 decimal place (#15759)

The OWM returns precipitation forecast values as they are submitted to their network. It can lead to values like `0.0025000000000004 mm` which does not make sense and messes up the display card. This PR rounds the value to 1 decimal place.

salbertson committed:
Add a "Reviewed by Hound" badge (#15767)

fabaff committed:
Upgrade Mastodon.py to 1.3.1 (#15766)

dgomes committed:
This component API has been decomissioned on the 31st of May 2018 by Telstra (#15757)

See #15668

fabaff committed:
Fix docstrings (#15770)

MatMaul committed:
Update pynetgear to 0.4.1 (bugfixes) (#15768)

miawgogo committed:
Opensky altitude (#15273)

* Added Altitude to opensky

* decided to take all metadata

* Final Tidy

* More formatting

* moving CONF_ALTITUDE to platform

* Moved CONF_ALTITUDE to platform

soraxas committed:
Expose internal states and fixed on/off state of Dyson Fans (#15716)

* exposing internal state and fixed onoff state

* fixed styling

* revert file mode changes

* removed self type hints

* better unit test and changed the way to return attributes

* made wolfie happy

Commit:
Setup fails when authentication fails.

Commit:
Setup fails when authentication fails.

MartinHjelmare commented on homeassistant/components/device_tracker/ritassist.py:
```diff
@@ -0,0 +1,89 @@
+class RitAssistDeviceScanner:
+    """Define a scanner for the RitAssist platform."""
+
+    def __init__(self, hass, config, see, discovery_info):
```
`hass` and `discovery_info` aren't used.

MartinHjelmare commented on homeassistant/components/device_tracker/ritassist.py:
```diff
@@ -0,0 +1,89 @@
+
+    def setup(self, hass):
+        """Setup a timer and start gathering devices."""
+        from homeassistant.helpers.event import track_utc_time_change
```
Move this to the other home assistant grouped imports at the top of the module.

MartinHjelmare commented on homeassistant/components/device_tracker/ritassist.py:
```diff
@@ -0,0 +1,89 @@
+
+    def _refresh(self) -> None:
+        """Refresh device information from the platform."""
+        import requests
```
Move this to 3rd party grouped imports (voluptuous) at the top of the module.

MartinHjelmare commented on homeassistant/components/device_tracker/ritassist.py:
```diff
@@ -0,0 +1,89 @@
+
+
+def setup_scanner(hass, config: dict, see, discovery_info=None):
+    """Validate the configuration and return a RitAssist scanner."""
```
We don't return scanner instance. We just create one.

MartinHjelmare commented:
Remember to run `script/gen_requirements.py` too. Otherwise the requirements file will not be correct.

Commit:
Move imports

Commit:
Move imports

depl0y commented:
@MartinHjelmare Running `script/gen_requirements_all.py` causes a lot of changes in `requirements_all.txt`, which are all downgrades or removals. I already tried bringing my fork up to date with:

```
git fetch upstream
git pull upstream dev
```

After this, still the same symptoms (downgrades + removals in the TXT files). am not sure I dare to push that to my repository.

MartinHjelmare commented:
I think you need to rebase on latest dev instead of merging. Since your branch is called the same as upstream dev, that is harder though. Always take for habit to create a new local feature branch with a different name than the upstream base remote branch. That will make things a lot easier and less error prone.

If it's hard to fix via command line, you can start over and copy your changes to a fresh branch. Then run the script to update requirements. Then open a new PR.

Commit:
Update requirements file

Commit:
Merge branch 'dev' of https://github.com/depl0y/home-assistant into dev

depl0y commented:
@MartinHjelmare A rebase fixed it in the end. But I am not sure where all the code review requests come from. This doesn't seem good, right?

depl0y commented:
Closing because of merge conflicts, opening a new PR
